packages/fiori-mcp-server/src/server.ts

```diff
@@ -95,8 +95,9 @@ export class FioriFunctionalityServer {
                 TelemetryHelper.markToolStartTime();
                 const telemetryProperties: TelemetryData = {
                     tool: name,
-                    functionalityId: (args as any)?.functionalityId
+                    functionalityId: (args as { functionalityId: string })?.functionalityId
```

Comment:
This cast is kind of odd and actually does the same as `any`. It doesn't even take into account that `functionalityId` is optional


```suggestion
                    functionalityId: ('functionalityId' in args) ? args.functionalityId : undefined
```